### PR TITLE
test: fix fs birth time test failure

### PR DESCRIPTION
**Pull request by vtjnash**
The claim here was false. Relax the test to claim something that is hopefully true.

Fixes: https://github.com/libuv/libuv/issues/2235
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/1694/

### CODE REVIEW

File: test/test-fs.c [hunk 1]
```diff
@@ -1409,13 +1411,14 @@ TEST_IMPL(fs_fstat) {
...
 
 #if defined(__linux__)
-  /* If statx() is supported, the birth time should be equal to the change time
-   * because we just created the file. On older kernels, it's set to zero.
-   */
+  /* If statx() is supported, the birth time should be recent and before the
+   * most recent change. On older kernels, it's set to zero. I've observed now
+   * being 1578590026.1812 and birth being 1578590025.996949689 in a short test
+   * run on Ubuntu 18.04.2 LTS Linux 4.15.0-72-generic x86_64, so we give a 10
+   * minute window to account for file-system clock skew. */
```
Comment (bnoordhuis) [inline test/test-fs.c]: With what file system is that? I checked xfs and ext4 and they both literally set the two fields to the same value (and honestly, I can't think of any other reasonable way you'd implement that.)

File: test/test-fs.c [hunk 2]
```diff
@@ -3714,7 +3717,7 @@ TEST_IMPL(fs_file_pos_after_op_with_offset) {
 }
 
 #ifdef _WIN32
-static void fs_file_pos_common() {
+static void fs_file_pos_common(void) {
```
Comment (bnoordhuis) [inline test/test-fs.c]: Can you split this out into a separate commit?

File: test/test-fs.c (hunk 1 above)
Comment (vtjnash) [reply test/test-fs.c]: Apparently this system was built using btrfs:
```
$ mount | grep /data
/dev/sda1 on /data type btrfs (rw,relatime,space_cache,subvolid=5,subvol=/)
```

Comment (vtjnash) [reply test/test-fs.c]: And, fwiw, we change the file a little while after creating it. So while the probably started the same, they aren't necessarily still the same when we called `fstat`.

Comment (bnoordhuis) [reply test/test-fs.c]: I'm having one of those "Oh... right..." moments right now. Can I suggest #2621 as an alternative?
